tornettools/tornettools

```diff
@@ -314,6 +307,18 @@ def main():
         action="store", dest="load_scale",
         default=1.0)
 
+    generate_parser.add_argument('--onion_service_user_scale',
+        help="""A float number by which we scale the number of onion-service users and servers."""
+        metavar="N", type=__type_fractional_float,
+        action="store", dest="onion_service_user_scale",
+        default=0.0)
+
+    generate_parser.add_argument('--exit_user_scale',
+        help="""A float number by which we scale the number of exit (non-onion-service) users and servers."""
+        metavar="N", type=__type_fractional_float,
+        action="store", dest="onion_service_user_scale",
+        default=1.0)
```

Comment:
More help details would be useful:
- Should we explain that it's a multiplicative factor?
- How does the user scale factors affect traffic load? Are the the total number of circuits the same but spread out over the higher number of users (so each user is "less busy")? Or is the circuits per user the same but the total number of circuits increases as a result of the new users?
 
If I'm reading the code correctly, increase these user scale factors implicitly increases total load (the global total number of circuits) by asking each tgen process to create the additional "users worth of circuits". Ultimately, I think that mean these `user_scale` factors are just another way to adjust the number of circuits that each tgen process ends up making but they have a very similar effect to using the load_scale option. (They do allow scaling onion service and exit traffic separately though, which is nice.)

Anyway, adding the details in the help message will help us remember what the hell we did later :) I've found the other (long) help messages here incredibly useful when generating networks.

Reply:
> Should we explain that it's a multiplicative factor?

Done

>  Or is the circuits per user the same but the total number of circuits increases as a result of the new users?

This one. Added a note here.